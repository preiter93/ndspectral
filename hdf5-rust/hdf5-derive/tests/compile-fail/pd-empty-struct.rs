extern crate hdf5_derive;
use hdf5_derive::H5Type;

use std::marker::PhantomData;

#[derive(H5Type)]
//~^ ERROR proc-macro derive
//~^^ HELP Cannot derive H5Type for empty structs
struct Foo<T> {
    t: PhantomData<T>,
}

fn main() {}
