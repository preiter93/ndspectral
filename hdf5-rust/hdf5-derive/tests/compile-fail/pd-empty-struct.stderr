error: proc-macro derive panicked
 --> $DIR/pd-empty-struct.rs:6:10
  |
6 | #[derive(H5Type)]
  |          ^^^^^^
  |
  = help: message: Cannot derive H5Type for empty structs
